### ISSUE TITLE
When restoring to a new course, template is not restored.

### ISSUE THREAD

**Issue by emmarichardson** [mdjnelson/moodle-mod_customcert#172]
If I restore a course with a custom certificate that has was created from a template, the template settings are not restored and the template has to be reapplied in the new course.

**Comment by trishevans** [comment]
Is this a known issue? We have experienced this issue today in Moodle version 3.4.3. Just curious to know if a fix has been incorporated into a later version of the plugin?

**Comment by Markibacsi** [comment]
I had this problem too!
My workaround was to not use the certificate template! Instead, I have created a course template - with the certificate module included. All restores (including bulk course creation) using this method are working fine.

My only problem now is that the teachers name needs to be manually applied. If nothing is changed the certificates are generated without the teacher's name. The assigned teacher's name appears in the element settings, but the element settings need to be accessed and saved for it to appear on the final certificate.

**Comment by trishevans** [comment]
Thanks for the advice, I will explore this as an option for us.

**Comment by emmarichardson** [comment]
I am finding that the background image still does not apply but I think some of the other pieces do come through.  Luckily(?) we use different images for each different amount of credit hours, so I have trained the staff to just apply the correct credit hour template as part of their course set up.

**Comment by trishevans** [comment]
Thanks Emma. Sounds like you have a good workaround. We don't make any adaptations so would prefer the background to restore. At this stage we are having manually update the template each time we restore a course which isn't ideal as it has the potential to be missed

**Comment by emmarichardson** [comment]
No, it is not a great solution...that is just on one site though...on other sites it is something that you have to remember to do...

**Comment by kattekrab** [comment]
Sadly, this one is probably not an easy fix. 

Am not sure if I’ll be able to get someone to look at this soon or not, sorry! :(

**Comment by jongseok** [comment]
Is this issue still in a backlog or is a developer working on it to fix the bug? Tx!

**Comment by markn86** [comment]
Sorry all.

I no longer work for Moodle HQ (though I am doing some contract work atm) and am focusing on advancing my career in other technologies which means less Moodle development time.

I would appreciate a patch for this and will review it in my spare time if someone was to supply it.

Cheers.

**Comment by mdjnelson** [comment]
Closing due to inactivity as part of issue housekeeping. I’m a solo maintainer with limited time right now, so I’m focusing on issues with recent activity and clear next steps.

If this is still relevant, please comment with an update (bug: current repro steps + environment; request: use case + desired outcome). With updated info, I can re-triage and reopen or track it in a new issue if needed.
